File: decompiled/sources/com/hjq/base/action/HandlerAction.java
package com.hjq.base.action;

import android.os.Handler;
import android.os.Looper;
import android.os.SystemClock;

public interface HandlerAction {
    public static final Handler HANDLER = new Handler(Looper.getMainLooper());

    Handler getHandler();

    boolean post(Runnable runnable);

    boolean postAtTime(Runnable runnable, long j);

    boolean postDelayed(Runnable runnable, long j);

    void removeCallbacks();

    void removeCallbacks(Runnable runnable);

    /* renamed from: com.hjq.base.action.HandlerAction$-CC  reason: invalid class name */
    public final /* synthetic */ class CC {
        public static Handler $default$getHandler(HandlerAction _this) {
            return HandlerAction.HANDLER;
        }

        public static boolean $default$post(HandlerAction _this, Runnable runnable) {
            return _this.postDelayed(runnable, 0);
        }

        public static boolean $default$postDelayed(HandlerAction _this, Runnable runnable, long j) {
            if (j < 0) {
                j = 0;
            }
            return _this.postAtTime(runnable, SystemClock.uptimeMillis() + j);
        }

        public static boolean $default$postAtTime(HandlerAction _this, Runnable runnable, long j) {
            return HandlerAction.HANDLER.postAtTime(runnable, _this, j);
        }

        public static void $default$removeCallbacks(HandlerAction _this, Runnable runnable) {
            HandlerAction.HANDLER.removeCallbacks(runnable);
        }

        public static void $default$removeCallbacks(HandlerAction _this) {
            HandlerAction.HANDLER.removeCallbacksAndMessages(_this);
        }
    }
}

File: decompiled/sources/com/hjq/base/action/HandlerAction.kt
package com.hjq.base.action

import android.os.Handler
import android.os.Looper
import android.os.SystemClock

interface HandlerAction {
    fun getHandler(): Handler {
        return HANDLER
    }

    fun post(runnable: Runnable): Boolean {
        return postDelayed(runnable, 0)
    }

    fun postDelayed(runnable: Runnable, delayMillis: Long): Boolean {
        val safeDelayMillis = if (delayMillis < 0) 0 else delayMillis
        return postAtTime(runnable, SystemClock.uptimeMillis() + safeDelayMillis)
    }

    fun postAtTime(runnable: Runnable, uptimeMillis: Long): Boolean {
        return HANDLER.postAtTime(runnable, this, uptimeMillis)
    }

    fun removeCallbacks(runnable: Runnable) {
        HANDLER.removeCallbacks(runnable)
    }

    fun removeCallbacks() {
        HANDLER.removeCallbacksAndMessages(this)
    }

    companion object {
        @JvmField
        val HANDLER: Handler = Handler(Looper.getMainLooper())
    }

    object CC {
        @JvmStatic
        fun `$default$getHandler`(handlerAction: HandlerAction): Handler {
            return HANDLER
        }

        @JvmStatic
        fun `$default$post`(handlerAction: HandlerAction, runnable: Runnable): Boolean {
            return handlerAction.postDelayed(runnable, 0)
        }

        @JvmStatic
        fun `$default$postDelayed`(
            handlerAction: HandlerAction,
            runnable: Runnable,
            delayMillis: Long
        ): Boolean {
            val safeDelayMillis = if (delayMillis < 0) 0 else delayMillis
            return handlerAction.postAtTime(runnable, SystemClock.uptimeMillis() + safeDelayMillis)
        }

        @JvmStatic
        fun `$default$postAtTime`(
            handlerAction: HandlerAction,
            runnable: Runnable,
            uptimeMillis: Long
        ): Boolean {
            return HANDLER.postAtTime(runnable, handlerAction, uptimeMillis)
        }

        @JvmStatic
        fun `$default$removeCallbacks`(handlerAction: HandlerAction, runnable: Runnable) {
            HANDLER.removeCallbacks(runnable)
        }

        @JvmStatic
        fun `$default$removeCallbacks`(handlerAction: HandlerAction) {
            HANDLER.removeCallbacksAndMessages(handlerAction)
        }
    }
}
